### PR TITLE
Add synthesis and translation-service documentation

### PR DESCRIPTION
**Pull request by ajskateboarder**
These are a couple of small internal APIs, but I think they are worth documenting for reverse-engineering purposes :)

### CODE REVIEW

File: api/synth_trans.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# `synthesis-service.` and `translate-service.` APIs
+
+These retrieve the speech synthesis and translation results for their respective extensions.
+
+### `GET translate-service.scratch.mit.edu/translate?language=<language code>&text=<text>`
+
+Retrieves the translation of the input text in the given language. Assumed to use Google Translate
```
Comment (towerofnix) [inline api/synth_trans.md:7]: I found [this citation](https://scratch.mit.edu/discuss/post/3778811/) for the translate extension using Google Translate. It's from four years ago (almost to the day!), but that's also a while after 3.0 released, so it's possibly still accurate. Maybe we could add something like this?

```suggestion
Retrieves the translation of the input text in the given language. Uses Google Translate [as of Dec. 9, 2019](https://scratch.mit.edu/discuss/post/3778811).
```

**Review by towerofnix — CHANGES_REQUESTED** [review]
Thanks! This looks great. I left some feedback with small suggestions - and a few bigger ones:

* According to [the Scratch Wiki](https://en.scratch-wiki.info/wiki/Translate_Extension), there's also an endpoint https://translate-service.scratch.mit.edu/supported. Could we add this to these docs? (On the wiki it includes `?lang=en` but I didn't see anything change when I tried some other language codes, or just removed the option).
  * This is apparently leftover from [the olden days](https://github.com/scratchfoundation/scratch-vm/blob/39d2559e37af4c384aca0c5c25bd871b0823eb0c/src/extensions/scratch3_translate/index.js#L69-L76) - it's seemingly not used by Scratch anymore. Check out @ericrosenbaum's pull request https://github.com/scratchfoundation/scratch-vm/pull/1159 which replaces it (and some discussion of the replacement at the bottom of this review).
* It would be cool to write what service the text-to-speech API uses too. According to [pretty much everyone](https://ocular.jeffalo.net/search?q=%22text%20to%20speech%22%20amazon&sort=relevance) ([including Jeffalo in 2021](https://scratch.mit.edu/discuss/post/5809755/)) it's Amazon Polly. Could we mention that? (The oldest source I've got is from [the wiki user Chikorita Lover](https://en.scratch-wiki.info/w/index.php?title=Extension&diff=217190&oldid=215050) around 3.0's released. Which, hey, [ceebee quoted on Wiki Wednesday literally two days ago, so...?](https://scratch.mit.edu/discuss/topic/728102/?page=1#post-7681434))
* Since the options here are unique and not necessarily self-descriptive, it would be nice to have some examples or descriptions. (For example, that `gender` accepts `male` and `female`.) I also noticed that you've named the two similar-looking options `<language code>` and `<locale>`, do those actually accept different kinds of values?
  * For text-to-speech, I'd strongly recommend linking to [this part of the extension's source code](https://github.com/scratchfoundation/scratch-vm/blob/489111f4d74909c2adac40ade1618f966ba30c34/src/extensions/scratch3_text2speech/index.js#L194-L341). The `speechSynthLocale` values are the ones that actually get provided to the API, as far as I can tell.
  * For translate, all the information is in [this NPM package](https://www.npmjs.com/package/scratch-translate-extension-languages?activeTab=code) (in `index.js`; this repo isn't on GitHub but its source is visible on NPM). There are a lot of dynamics here to do with actually downloading a supported langauge list from Google, apparently. I'm not versed enough to understand it properly though! There's seemingly a full, pre-built list of languages in `language.json` here too, which is what the extension [uses](https://github.com/scratchfoundation/scratch-vm/blob/develop/src/extensions/scratch3_translate/index.js#L6) (see `package.json`). So the list of locales is the keys in that file, I guess.

File: api/synth_trans.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+### `GET synthesis-service.scratch.mit.edu/synth?locale=<locale>&gender=<gender>&text=<text>`
+
+Retrieves the synthesis of the input text as an audio file
```
[the remark this answers is not in the document]
Comment (ajskateboarder) [reply api/synth_trans.md]: Fixed this 👍

File: api/synth_trans.md (hunk 1 above)
Comment (ajskateboarder) [reply api/synth_trans.md]: Added 👍 (idk what else to say lol)

**Comment by ajskateboarder** [comment]
> I also noticed that you've named the two similar-looking options <language code> and <locale>, do those actually accept different kinds of values?

It seems the \<locale> parameter is for written languages (e.g. `en`, `hi`, `ja`) while the \<language code> param is for spoken language (e.g. `en-US`, `hi-IN`, `ja-JP`). I'll reword them to be less ambiguous, maybe use `locale` and `speech locale`

**Comment by ajskateboarder** [comment]
@towerofnix Bump on the review 👀 

I made changes based on what you have suggested

**Review by towerofnix — APPROVED** [review]
I made some tweaks for clarity/formatting and filling out some slightly more particular details on the parameter options - this is looking good to me now, but I'd like to get a reply/LGTM from you since I've made changes! Thanks for your work on this PR.

**Comment by ajskateboarder** [comment]
Thanks for your edits. They definitely added more clarity, lgtm ^^

**Comment by towerofnix** [comment]
Merged - thank you!

**Pull request merged**